src/Symfony/Component/DependencyInjection/EnvVarProcessor.php

```diff
@@ -85,71 +85,66 @@ public function getEnv($prefix, $name, \Closure $getEnv)
             throw new RuntimeException(sprintf('Non-scalar env var "%s" cannot be cast to %s.', $name, $prefix));
         }
 
-        if ('string' === $prefix) {
-            return (string) $env;
-        }
+        switch ($prefix) {
```

Comment:
I recommend to add a help comment here to avoid some people in the future refactoring it into some `if` to "improve code quality". Something like this:

```php
// used a 'switch' instead of 'if' conditions to enable a PHP micro-optimization
```